sqflite integration for Flutter (#6500)
getsentry/sentry-docs: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx

### File I/O Instrumentation

> The Sentry-specific file I/O implementation starts a span out of the active transaction, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`, `file.write`, `file.delete`, `file.open`, `file.read` or `file.rename`, and `description` to `filename` (for example, `file.txt`).

Reviewer: ```suggestion
The Sentry-specific file I/O instrumentation starts a span out of the active transaction, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`, `file.write`, `file.delete`, `file.open`, `file.read` or `file.rename`, and `description` to `filename` (for example, `file.txt`).
```

> The Sentry-specific file I/O implementation starts a span out of the active transaction, bound to the scope for each file I/O operation. The SDK sets the span `operation` to `file.copy`, `file.write`, `file.delete`, `file.open`, `file.read` or `file.rename`, and `description` to `filename` (for example, `file.txt`).

Reviewer 2: ```suggestion
The Sentry-specific file I/O instrumentation starts a span from an active transaction that's bound to the scope of each file I/O operation. The SDK sets the span `operation` to `file.copy`, `file.write`, `file.delete`, `file.open`, `file.read` or `file.rename` and the `description` to `filename` (for example, `file.txt`).
```

Author: This text has not changed, it was just a typo from `implementation` to `instrumentation`.
We use the "out of the active transaction" in multiple places, I suggest doing it in a separate PR.

> In addition, the span contains other useful information such as `file.size` (raw number of bytes), `file.path` (an absolute path to the file) and `file.async` (`true` if the called method returns a `Future`, or `false` if it's a `Sync` call) as part of the `data` payload.

Reviewer 2: ```suggestion
In addition, the span contains other useful information as part of the `data` payload, such as: `file.size`, (the raw number of bytes), `file.path`, (an absolute path to the file), and `file.async`, (which is `true` if the called method returns a `Future`, or `false` if it's a `Sync` call).
```

Author: This text has not changed, it was just a typo from `span ` to `transaction `.
Can we do it in a different PR?

> When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).

Reviewer 2: ```suggestion
When the operation throws an `Exception`, Sentry's SDK associates it with the running span. If you haven't set the SDK to swallow and capture the exception, the span and `SentryEvent` will be shown as linked on the **Issue Details** page in [sentry.io](https://sentry.io).
```

Author: Same here, this text is in multiple places and has not been changed.

Learn more about our [file I/O integration](/platforms/dart/configuration/integrations/file/).

### sqflite Database Instrumentation

> The sqflite Database Instrumentation starts a span out of the active transaction, bound to the scope for each `CRUD` operation.

Reviewer: ```suggestion
The sqflite database instrumentation starts a span out of the active transaction, bound to the scope for each `CRUD` operation.
```

> The sqflite Database Instrumentation starts a span out of the active transaction, bound to the scope for each `CRUD` operation.

Reviewer 2: ```suggestion
The sqflite database instrumentation starts a span from an active transaction that's bound to the scope of each `CRUD` operation.
```

Author: Replaced the `out of the active transaction...` to from an active transaction...` everywhere.

> 
> The span finishes once the operation has been executed. The span `status` is set to `SpanStatus.ok` if successful or `SpanStatus.internalError` if there was any error.
> 
> When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the **Issue Details** page in [sentry.io](https://sentry.io).
> 
> Learn more about our [sqflite Database Instrumentation](/platforms/flutter/configuration/integrations/sqflite-instrumentation/).

Reviewer 2: ```suggestion

The span finishes once the operation has been executed. The span `status` is then set to `SpanStatus.ok` if successful, or `SpanStatus.internalError` if there was an error.

When the operation throws an `Exception`, Sentry's SDK associates it with the running span. If you haven't set the SDK to swallow and capture the exception, the span and `SentryEvent` will be shown as linked on the **Issue Details** page in [sentry.io](https://sentry.io).

Learn more about our [sqflite Database Instrumentation](/platforms/flutter/configuration/integrations/sqflite-instrumentation/).
```
